src/dockersync.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+DOCKER_ROOT=${DOCKER_ROOT:-/var/lib/docker}
+BTRFS_ROOT=${BTRFS_ROOT:-$DOCKER_ROOT/btrfs/subvolumes}
+
+src=$1
+dest=$2
+config=$3
+
+src_id=$(docker inspect -f '{{ .Id }}' "$src")
+dest_id=$(cat /app/empty.tar | docker import -)
+
+jq ".config=$config" "$DOCKER_ROOT/graph/$dest_id/json" > "$DOCKER_ROOT/graph/$dest_id/json.tmp"
+mv "$DOCKER_ROOT/graph/$dest_id/json.tmp" "$DOCKER_ROOT/graph/$dest_id/json"
+
+btrfs subvolume delete "$BTRFS_ROOT/$dest_id"
+btrfs subvolume snapshot "$BTRFS_ROOT/$src_id" "$BTRFS_ROOT/$dest_id"
+
+rsync --timeout=300 --archive --delete --read-batch=- "$BTRFS_ROOT/$dest_id"
+
+sync
+
+docker tag -f "$dest_id" "$dest"
```

Comment:
Can the tagging be handled by the host docker via returning these to the supervisor and then calling dockerode's tag?